dmoj/cptbox/helper.cpp

```diff
@@ -162,21 +163,25 @@ int cptbox_child_run(const struct child_config *config) {
 
         unsigned int child_arch = get_seccomp_arch(config->debugger_type);
 
+        int rc;
         if (seccomp_arch_exist(ctx, child_arch) == -EEXIST &&
-            seccomp_arch_add(ctx, child_arch) != 0) {
-            perror("seccomp_arch_add");
+            (rc = seccomp_arch_add(ctx, child_arch)) != 0) {
+            fprintf(stderr, "seccomp_arch_add: %s\n", strerror(-rc));
             return 203;
         }
 
         for (int syscall = 0; syscall < MAX_SYSCALL; syscall++) {
             if (config->syscall_whitelist[syscall]) {
-                seccomp_rule_add(ctx, SCMP_ACT_ALLOW, syscall, 0);
+                if (rc = seccomp_rule_add(ctx, SCMP_ACT_ALLOW, syscall, 0)) {
+                    fprintf(stderr, "seccomp_rule_add(..., %d): %s\n", syscall, strerror(-rc));
```

Comment:
I suppose it makes no difference if this failed, since it will just end up trapping ptrace. Separate PR?

Reply:
Added explanatory comment, but we should keep the log.